include/swift/SIL/GraphOperationInfo.h

```diff
@@ -145,18 +145,27 @@ struct GraphOperationInfo {
       assert(getKind() == SOK_List);
       return OperandList;
     }
+
+    /// Returns the result of GraphOperationInfo::decodeOperandName on this
+    /// operand.
+    std::pair<StringRef, OperandLowering> decodeName() const;
```

Comment:
This API and the next 3 APIs confused me a bit -- then I realized this API belonged to a different struct/class! :-)

To reduce chances of confusion, can we rename this API to sth like decodeAttrName, and rename getName() below to sth like getOpName()? 

decodeName() below can also use some refinement, say decode(). But I'm less concerned with that one.

Reply:
How about:
* rename "StructuredOperand" to "StructuredArgument" because "Operand" and "Operation" look very similar and so it's very easy to confuse APIs like `getOperandName` with `getOperationName`. (also rename all the other classes and variables related to "StructuredOperand" that have "Operand" in their name)
* change this api to `getArgumentNameAndLowering()` the fact that it's "decoding" is less relevant than the fact that it's returning name and lowering. also I'm using "argument" instead of "attribute" because these things are also allowed to be inputs now.
* rename `getName()` below to `getOperationName()`

I have done these things in the code.

Reply:
SG. Are "arguments" basically "attrs" -- if yes consider adding a comment to make that association.

a tfop (graph_op) can also have input tensors as "arguments", but they are of a different kind and need not be decoded / demangled here.